react/QuotaAlert/index.jsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import get from 'lodash/get'
+import Modal from '../Modal'
+import { withClient } from 'cozy-client'
+import { isMobileApp } from 'cozy-device-helper'
+
+import useInstance from '../../helpers/useInstance'
+
+import withLocales from '../I18n/withLocales'
+import en from './locales/en.json'
+import fr from './locales/fr.json'
+
+const locales = {
+  en,
+  fr
+}
+
+const buildPremiumLink = (uuid, managerUrl) =>
+  `${managerUrl}/cozy/instances/${uuid}/premium`
+
+export const QuotaAlert = ({ t, onClose, client }) => {
+  let uuid, managerUrl
+  /**
+   * We do the request only on the web since Apple
+   * and Google have a restricted policy for the
+   * inApp purchase...
+   */
+  if (!isMobileApp()) {
+    const instanceInfo = useInstance(client)
```

Comment:
I'm no hooks expert, but it looks like we are breaking this rule here: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Are we sure we don't introduce potential bugs?

Reply:
Yep :+1:

Reply:
I think there is no bug here but it is reliant on the fact that isMobileApp does not change. If it were to change, we would potentially have a bug here.